tests/test_bt_merge.cpp

```diff
@@ -1,12 +1,76 @@
+#include <rapidcheck.h>
+
 #include <catch2/catch_test_macros.hpp>
 
 #include "catch2_bt_format.hpp"
+#include "rapidcheck/catch.h"
 #include "session/bt_merge.hpp"
 
 using oxenc::bt_dict;
 using oxenc::bt_list;
 using oxenc::bt_value;
 
+namespace rc {
+
+template <>
+class Arbitrary<std::string_view> {
+  public:
+    static rc::Gen<std::string_view> arbitrary() {
+
+        return rc::gen::map(rc::gen::arbitrary<std::string>(), [](const std::string& s) {
+            return std::string_view(s);
+        });
+    }
+};
+
+Gen<bt_value> genBtValue() {
+    return gen::oneOf<bt_value>(
+            gen::map(
+                    gen::arbitrary<std::string>(),
+                    [](const std::string& s) { return bt_value{s}; }),
+            gen::map(
+                    gen::arbitrary<std::string_view>(),
+                    [](const std::string_view& s) { return bt_value{s}; }),
+            gen::map(gen::arbitrary<int64_t>(), [](int64_t i) { return bt_value{i}; }),
+            gen::map(gen::arbitrary<uint64_t>(), [](uint64_t i) { return bt_value{i}; }),
+            gen::map(
+                    gen::resize(5, gen::container<bt_list>(gen::lazy(&genBtValue))),
+                    [](const bt_list& list) { return bt_value{list}; }),
+            gen::map(
+                    gen::resize(
+                            5,
```

Comment:
another hard coded length

Reply:
`random % upperbound`

Reply:
I actually come up with a more interesting solution.
RapidCheck has a `gen::scale()` function which is aware of the size of the current generator, and scale it by a ratio before making another generator.

https://github.com/emil-e/rapidcheck/blob/master/doc/generators_ref.md#gent-scaledouble-scale-gent-gen

By using `gen::scale(0.5, )`, we are able to create complex structures which are both wide and deep, while the generator still guarantee to converge, terminate, without running out of memeory.

See https://gist.githubusercontent.com/fracting/1f3a9d12cbdc896b1e472167459e2801/raw/7b2c117f072fe4f77f42f03a28a1882099b032d9/bt_dict%2520test%2520case for one example generated by RapidCheck.